### DISCUSSION TITLE
Defining a Type from TypeBox and preserve tags

### DISCUSSION THREAD

**Discussion by gonzauy** [sinclairzx81/typebox#850] [Q&A] [answered]
Hi 👋 ,

Based on this example:

```
export const SchemaTest = Type.Object(
  {
    name: Type.String({ description: 'The name of the user' }),
    age: Type.Number({
      minimum: 18,
      maximum: 100,
      description: 'The age of the user, must be at least 18',
    }),
  },
);

export type MyType = Static<typeof SchemaTest>;
```

Is there a way to conserve the tags from the TypeBox `SchemaTest` to the type `MyType`?

Would like to preserve the information defined on the schema like the `description, min, max` of each properties as `JSDoc` on the type `MyType`.

Thank you! 🙏

**Comment by sinclairzx81** [comment, marked as answer]
@gonzauy Hello!

I don't know of a way to map schema options to JsDoc comments in the type system, but you can apply JsDoc comments manually and have them come through on the static type.

[TypeScript Link Here](https://www.typescriptlang.org/play/?#code/JYWwDg9gTgLgBAbzgFQJ5gKYBo4GUYCGMwAxnAL5wBmUEIcA5AAIDOwAdiQDYHBQD0MdBgBGEAB4MAUFIzjIsOCQjsW8XCQAWGEAWQY1cALwphAOgDyIgFYYSMABRS4iZy7j8AVJ7hv3P5G04dgIQDDgIKjgYIIBXFgwoPxdPfmSQsIAuU0wzfCgOAHMHJAATAxICsGIVbIZA8IzwyOi4hKgGCgBKLGSvH2S4AKCCQuaomPD4xJwQePgRcKI4LgwCQwBGAA5fdxS4JhAOUFj6bcGfQ4JxE7OABjuLtL3RjGy0XIA5U8WoEsG4Ed2LdsttensXLobnMQKCHuCIeUWJVgNVgLVGA04K8IhM2jNAfM4ItsfBVut4NsGAiXOQem5yL0ugBuGRyBTwISYOAAWVQH3CJnwRFIAB4uRgWhptLp9GoAHysqRUWKcGrsaIGRwANwIXFib15-OEXVcLl1+owZiaHn4cE0EG1iTcFoNZhxtvtjud5CAA)

```typescript
import { Type, Static } from '@sinclair/typebox'

export const SchemaTest = Type.Object(
  {
    /** 
     * The name of the user
     */
    name: Type.String({ description: 'The name of the user' }),
    /** 
     * The age of the user, must be at least 18 
     * @minimum 18
     * @maximum 100
     */
    age: Type.Number({
      minimum: 18,
      maximum: 100,
      description: 'The age of the user, must be at least 18',
    }),
  },
);

export type MyType = Static<typeof SchemaTest>;

function test(value: MyType) {
  value.name // hover
  value.age  // hover
}
```
Unfortunately, this is a bit of duplication, but should largely work. I would be super keen to explore comment mapping should it become possible in the language, but sadly not possible today (afaik)

Hope this helps
S

**Comment by gonzauy** [comment]
@sinclairzx81 , thank you for the answer!

**Comment by castarco** [comment]
@gonzauy I just published https://www.npmjs.com/package/@coderspirit/nominal-typebox .

For now it only preserves "brands", but I'll be iterating on it to also preserve arbitrary properties (it's based on `@coderspirit/nominal`, which allows to preserve arbitrary properties/annotations in our types).

As a side note, there is already one "property" that can be already preserved:
- that the number is an integer and not just an arbitrary number.
